xla/xla.proto

```diff
@@ -786,7 +786,12 @@ message DebugOptions {
   // a deterministic implementation.
   bool xla_gpu_exclude_nondeterministic_ops = 297;
 
-  // Next id: 300
+  // Enables temp User Buffer Registration. Enable this flag will use a 
+  // separate cuda async memory allocator to allocate temp buffer, this will 
+  // allocate temp buffer to the fix address on every iteration.
+  bool xla_gpu_enable_temp_async_buffer = 301;
+
+  // Next id: 302
```

Comment:
I think that I roughly understand this but the wording is not really clear to me.
IIUC the goal is to allocate a temp buffer which stays the same on every iteration? Why is it important that the buffer is "async"? 

Shouldn't we call this flag something that emphasizes the stable nature of the buffer?

Reply:
I changed the flag name to xla_gpu_temp_buffer_use_cuda_async_allocator, it requires to use the `cudaAsyncAllocator` because it supports using separate memory pool for allocation, which will just ensure that the allocated address happens to be the stable.